api/controller.py

```diff
@@ -821,18 +834,40 @@ def crawlable_list_feed(self, list_name):
         )
         lane = CrawlableCustomListBasedLane()
         lane.initialize(library, list)
-        return self._crawlable_feed(library, title, url, lane)
-
-    def _crawlable_feed(self, library, title, url, lane, annotator=None):
-        annotator = annotator or self.manager.annotator(lane)
-        facets = CrawlableFacets.default(library)
-        pagination = load_pagination_from_request()
+        return self._crawlable_feed(title=title, url=url, lane=lane)
+
+    def _crawlable_feed(self, title, url, lane, annotator=None,
+                        feed_class=AcquisitionFeed):
+        """Helper method to create a crawlable feed.
+
+        :param title: The title to use for the feed.
+        :param url: The URL from which the feed will be served.
+        :param lane: A crawlable Lane which controls which works show up
+            in the feed.
+        :param annotator: A custom Annotator to use when generating the feed.
+        :param feed_class: A drop-in replacement for AcquisitionFeed
+            for use in tests.
+        """
+        pagination = load_pagination_from_request(
+            default_size=Pagination.DEFAULT_CRAWLABLE_SIZE
+        )
         if isinstance(pagination, ProblemDetail):
             return pagination
-        feed = AcquisitionFeed.page(
-            self._db, title, url, lane, annotator=annotator,
-            facets=facets,
-            pagination=pagination,
+
+        search_engine = self.search_engine
+        if isinstance(search_engine, ProblemDetail):
+            return search_engine
+
+        annotator = annotator or self.manager.annotator(lane)
+
+        # A crawlable feed has only one possible set of Facets,
+        # so library settings are irrelevant.
+        facets = CrawlableFacets.default(None)
+
+        feed = feed_class.page(
+            _db=self._db, title=title, url=url, lane=lane, annotator=annotator,
+            facets=facets, pagination=pagination, cache_type="crawlable",
```

Comment:
These used to be 'page' feeds, and it's not technically necessary for crawlable feeds to have a distinctive type, because they have a distinctive sort order, but it was really useful to be able to just delete all the crawlable feeds from the database while testing against a real site.

Reply:
This seems fine but probably needs some changes to CachedFeed in the model.

Reply:
I took a look at core and here are the changes we might want to make:

* In core, we might want to add a constant for 'crawlable' so other people know not to use that string.
* The Crawlable*Lane classes ought to define MAX_CACHE_AGE. Previously they were using the site's default cache time for nongrouped page feeds, and now the default cache time is zero. What's a good cache time for crawlable feeds?

Reply:
For the list feeds, I think the import script only checks once a day so the cache time could probably be a few hours. For a shared collection feed it checks every 6 hours and should probably be shorter.